code/Controllers/CMSMain.php

```diff
@@ -828,6 +833,44 @@ public function getEditForm($id = null, $fields = null)
         }
     }
 
+    protected function getArchiveWarningMessage($record)
+    {
+        // Get all page's descendants
+        $record->collateDescendants(true, $descendants);
+        if (!$descendants) {
+            $descendants = [];
+        }
+
+        // Get all campaigns that the page and its descendants belong to
+        $inChangeSetIDs = ChangeSetItem::get_for_object($record)->column('ChangeSetID');
+
+        foreach ($descendants as $page) {
+            $inChangeSetIDs = array_merge($inChangeSetIDs, ChangeSetItem::get_for_object($page)->column('ChangeSetID'));
+        }
+
+        if (count($inChangeSetIDs) > 0) {
+            $inChangeSets = ChangeSet::get()->filter(['ID' => $inChangeSetIDs, 'State' => ChangeSet::STATE_OPEN]);
+        } else {
+            $inChangeSets = new ArrayList();
+        }
+
+        $numCampaigns = ChangeSet::singleton()->i18n_pluralise($inChangeSets->count());
+        $numCampaigns = mb_strtolower($numCampaigns);
+
+        if (count($descendants) > 0 && $inChangeSets->count() > 0) {
+            $archiveWarningMsg = _t('CMSMain.ArchiveWarningWithChildrenAndCampaigns', 'Warning: This page and all of its child pages will be unpublished and automatically removed from their associated {NumCampaigns} before being sent to the archive.\n\nAre you sure you want to proceed?', [ 'NumCampaigns' => $numCampaigns ]);
+        } else if (count($descendants) > 0) {
```

Comment:
Please be careful of rebasing your work, I think you force pushed my CSLint fixes for updating `else if` to be `elseif` ;)